### ISSUE TITLE
Incomplete pagination support

### ISSUE THREAD

**Issue by rclement** [Mangopay/mangopay2-python-sdk#164]
The Python MangoPay SDK ignores MangoPay API pagination information specified in the documentation: https://docs.mangopay.com/guide/lists-pagination-management

For now, the only thing offered by the library is to retrieve a paginated list with a given page number and a number of items per page, such as:

```
>>> users = User.all(page=1, per_page=10)
[ { ... }, ... ]
```

However there is no way to get the total number of items and the total number of pages.
The library should make available at least these 2 informations for the requested list.
The following MangoPay API response headers contains those informations:

`X-Number-Of-Pages: 2`
`X-Number-Of-Items: 12`

At the moment, lacking those 2 informations makes the pagination support almost unusable (or maybe I missed something, if so please let me know).

Unfortunately, I'm a bit short of proposal but at first glance there is no quick-fix.
One potential way to integrate this would be return some metadata along the requested list, introducing a different `all()` method in order to maintain backward-compatibility with all `all()` method calls:

```
>>> users = User.all_paginated(page=1, per_page=10)
{
  num_items: 12,
  num_pages: 2,
  per_page: 10,
  page: 1,
  items: [ { ... }, ... ],
}
```

For reference, here are some other libraries takes on the matter:
- Flask-SQLAlchemy: http://flask-sqlalchemy.pocoo.org/2.3/api/#flask_sqlalchemy.Pagination
- Stripe: https://stripe.com/docs/api/python#pagination

What's your take on this?

**Comment by mickaelpois** [comment]
@rclement Hello and thanks for your feedback! I really apologize for the delay.

We will have a look at it, we must support it to provide you all the tools you need to manage your pagination well.

I will have a look and keep you posted very soon.

**Comment by rclement** [comment]
@mickaelpois Thanks for the positive answer!

Please do not hesitate to ask for some more developer feedback when evaluating various solutions for this feature, I'd be happy to help achieving the most elegant implementation (and this is not trivial).

**Comment by TamasNo1** [comment]
@mickaelpois do you have any update on this? Maybe an ETA for the fix?
We have tried to upgrade from the 2.x version of Mangopay SDK in order to have AVS support, but not being able to offer proper pagination support is very disappointing.

**Comment by mickaelpois** [comment]
Fixed by #178 on 3.6.0 version
